service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java

```diff
@@ -0,0 +1,218 @@
+     */
+    public <T> Observable<ServiceResponseWithHeaders<T, DigitalTwinGetHeaders>> getDigitalTwinWithResponse (@NonNull String digitalTwinId, Class<T> clazz)
+    {
+        return digitalTwin.getDigitalTwinWithServiceResponseAsync(digitalTwinId)
+                .map(response -> {
+                    T genericResponse = null;
+                    try {
+                        genericResponse = DeserializationHelpers.castObject(objectMapper, response.body(), clazz);
+                    } catch (JsonProcessingException e) {
+                        Observable.error(e);
+                    }
+                    return new ServiceResponseWithHeaders<>(genericResponse, response.headers(), response.response());
+                })
+                .subscribeOn(Schedulers.io());
+    }
+
+    /**
+     * Updates a digital twin.
+     * @param digitalTwinId The Id of the digital twin.
+     * @param digitalTwinUpdateOperations The JSON patch to apply to the specified digital twin. This argument can be created using {@link UpdateOperationUtility}.
+     * @return A {@link ServiceResponseWithHeaders} with {@link DigitalTwinUpdateHeaders}.
+     */
+    public Observable<ServiceResponseWithHeaders<Void, DigitalTwinUpdateHeaders>> updateDigitalTwinWithResponse (@NonNull String digitalTwinId, @NonNull List<Object> digitalTwinUpdateOperations)
+    {
+        return digitalTwin.updateDigitalTwinWithServiceResponseAsync(digitalTwinId, digitalTwinUpdateOperations)
+                .subscribeOn(Schedulers.io());
+    }
+
+    /**
+     * Updates a digital twin.
+     * @param digitalTwinId The Id of the digital twin.
+     * @param digitalTwinUpdateOperations The JSON patch to apply to the specified digital twin. This argument can be created using {@link UpdateOperationUtility}.
+     * @param options The optional settings for this request.
+     * @return A {@link ServiceResponseWithHeaders} with {@link DigitalTwinUpdateHeaders}.
+     */
+    public Observable<ServiceResponseWithHeaders<Void, DigitalTwinUpdateHeaders>> updateDigitalTwinWithResponse (@NonNull String digitalTwinId, @NonNull List<Object> digitalTwinUpdateOperations, @NonNull DigitalTwinUpdateRequestOptions options)
+    {
+        String ifMatch = options != null ? options.getIfMatch() : null;
+        return digitalTwin.updateDigitalTwinWithServiceResponseAsync(digitalTwinId, digitalTwinUpdateOperations, ifMatch)
+                .subscribeOn(Schedulers.io());
+    }
+
+    /**
+     * Invoke a command on a digital twin.
+     * @param digitalTwinId The Id of the digital twin.
+     * @param commandName The command to be invoked.
+     * @param payload The command payload.
+     * @return The application/json command invocation response.
+     */
+    public Observable<String> invokeCommand (@NonNull String digitalTwinId, @NonNull String commandName, String payload)
```

Comment:
"String payload" needs to be @NonNull, there needs to be another API overload without it

Reply:
1. @NonNull String digitalTwinId, @NonNull String commandName
2. @NonNull String digitalTwinId, @NonNull String commandName, @NonNull String payload
3. @NonNull String digitalTwinId, @NonNull String commandName, @NonNull String payload, @NonNull DigitalTwinInvokeCommandRequestOptions options => with http response